src/main/java/com/verily/genomewarp/GenomeWarpSerial.java

```diff
@@ -258,6 +264,107 @@ private static void warn(String message) {
     return matches;
   }
 
+  // TODO: docs
+  private static List<GenomeRange> generateBEDFromVCF(VCFFileReader vcfReader) {
+    List<GenomeRange> toReturn = new ArrayList<>();
+    for (final VariantContext var: vcfReader) {
+      // TODO: check that start and end positions are valid
+      // htsjdk start/end are both 1-based closed
+      GenomeRange curr = new GenomeRange(var.getChr(), var.getStart() - 6, var.getEnd() + 5);
+      toReturn.add(curr);
+    }
+    return toReturn;
+  }
+
+  private static void extendWithSplittedRegion(List<GenomeRange> regions, GenomeRange toAdd) {
+    long pos = toAdd.getStart();
+    int windowSize = ARGS.bedWindowSize;
+    for (; pos + windowSize <= toAdd.getEnd(); pos += windowSize) {
+      regions.add(new GenomeRange(toAdd.getChromosome(), pos, pos + windowSize));
+    }
+    if (pos != toAdd.getEnd()) {
+      regions.add(new GenomeRange(toAdd.getChromosome(), pos, toAdd.getEnd()));
+    }
+  }
+
+  // TODO: docs
+  private static List<GenomeRange> mergeRegionsFromQueryBEDAndVariants(
+      List<GenomeRange> queryBED, List<GenomeRange> fromVcfBED) {
+    List<GenomeRange> mergedBed = new ArrayList<>();
+    ListIterator<GenomeRange> queryIt = queryBED.listIterator();
+    ListIterator<GenomeRange> vcfIt = fromVcfBED.listIterator();
+    GenomeRange queryRegion = null;
+    GenomeRange vcfRegion = null;
+    boolean addVcf = true;
+    while (queryIt.hasNext() && vcfIt.hasNext()) {
```

Comment:
The logic of this merge is very similar to that in https://github.com/google/deepvariant/blob/r0.7/deepvariant/postprocess_variants.py#L745-L804 which I think is a bit clearer in intent with fewer conditionals, please consider refactoring.